python/lsst/daf/butler/_butler.py

```diff
@@ -42,15 +42,16 @@
 from ._butler_repo_index import ButlerRepoIndex
 from ._config import Config, ConfigSubset
 from ._dataset_existence import DatasetExistence
-from ._dataset_ref import DatasetIdGenEnum, DatasetRef
+from ._dataset_ref import DatasetId, DatasetIdGenEnum, DatasetRef
 from ._dataset_type import DatasetType
 from ._deferredDatasetHandle import DeferredDatasetHandle
 from ._file_dataset import FileDataset
 from ._limited_butler import LimitedButler
 from ._storage_class import StorageClass
+from ._timespan import Timespan
 from .datastore import DatasetRefURIs, Datastore
 from .dimensions import DataId, DimensionConfig
-from .registry import Registry, RegistryConfig, _RegistryFactory
+from .registry import CollectionArgType, Registry, RegistryConfig, _RegistryFactory
```

Comment:
We should probably move `CollectionArgType` out of `registry`. I think ideally `Butler` should not depend on anything from `registry` (we keep `Butler.makeRepo()` so we need three other classes here.)

Reply:
I replaced it with an explicit `str | Sequence[str]` definition.